src/base/memory_monitor_sr.cpp

```diff
@@ -20,18 +20,58 @@
  * memory_monitor_sr.cpp - Implementation of memory monitor module
  */
 
+#include <malloc.h>
+#include <cassert>
 #include <cstring>
 #include <algorithm>
-#include <cassert>
 
 #include "memory_monitor_sr.hpp"
 
+#ifndef HAVE_USR_INCLUDE_MALLOC_H
+#define HAVE_USR_INCLUDE_MALLOC_H
+#endif
+
+typedef struct mmon_metainfo MMON_METAINFO;
+struct mmon_metainfo
+{
+  uint64_t allocated_size;
+  int tag_id;
+  int magic_number;
+};
+
 namespace cubmem
 {
+  static const std::string magic_string ("MMON");
+  memory_monitor *mmon_Gl = nullptr;
+
   memory_monitor::memory_monitor (const char *server_name)
-    : m_server_name {server_name} {}
+    : m_server_name {server_name}
+  {
+    memcpy (&m_magic_number, magic_string.c_str (), sizeof (int));
```

Comment:
How about just setting `magic` to a random integer like `12345678` if you want it to be an integer? Then, it can be `static` easily without this copy. Or, tell me the point of the human-readable magic string "MMON" if I'm missing.

Reply:
I wish the magic number wasn't so easy, and it would be nice if it indicated where it's being used. Since this operation occurs only once when the memory_monitor is initially created, I don't think it's a major issue. If the way of setting this magic number is significantly flawed in terms of implementation, functionality, or performance, I'd appreciate an explanation. Otherwise, I don't feel strongly about changing it.

Reply:
I don't think the performance is a thing here. I think I haven't described my intention. From https://github.com/CUBRID/cubrid/pull/4918#discussion_r1472299166, what I'm trying to do is, I insist is, the `magic` or `m_magic_number` should be "constant". This must not be changed through the entire time, so I'd like to prevent it at all. And, the way I see it, it could be `static` as well.

In the previous comment, I agree that it would be costly if it's string. So, this time, I'm suggesting a way to keep it an integer.

> I wish the magic number wasn't so easy, and it would be nice if it indicated where it's being used.

Agreed. If I were you, I'd want to do so. But, it doesn't seem necessary and if we can resolve the issue I described.

Reply:
I agree about the magic number should be "constant". But I don't think the magic number should be "static". I think "static" is optional property of the magic number.
Then, how about making `m_magic_number` just a "constant" variable?
`const int m_magic_number = *reinterpret_cast<const int*> (magic_string.c_str ())`